### PR TITLE
WEA-12 Conversion of temperature unit

### PR DESCRIPTION
**Pull request by ayush-shta**
Also used shared preferences to store unit format.
@subash-poudel

### CODE REVIEW

File: app/src/main/java/com/ayush/weatherapp/WeatherApplication.java [hunk 1]
```diff
@@ -13,5 +13,7 @@
     if (BuildConfig.DEBUG) {
       Timber.plant(new DebugTree());
     }
+
+    PreferenceRepositoryImpl.init(getApplicationContext());
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/WeatherApplication.java]: this

File: app/src/main/java/com/ayush/weatherapp/constants/Temperature.java [hunk 2]
```diff
@@ -4,14 +4,14 @@
...
   private static final int CELSIUS = 0;
   private static final int FAHRENHEIT = 1;
 
   @IntDef({CELSIUS, FAHRENHEIT})
   @Retention(RetentionPolicy.SOURCE)
-  public @interface Temperature {
-    int CELSIUS = TemperatureConstant.CELSIUS;
-    int FAHRENHEIT = TemperatureConstant.FAHRENHEIT;
+  public @interface Unit {
+    int CELSIUS = com.ayush.weatherapp.constants.Temperature.CELSIUS;
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/constants/Temperature.java]: ??

File: app/src/main/java/com/ayush/weatherapp/customViews/TemperatureTextView.java [hunk 3]
```diff
@@ -30,41 +35,61 @@ public TemperatureTextView(Context context, AttributeSet attrs, int defStyleAttr
...
+  public void setTemperatureType(@Temperature.Unit int temperatureType) {
     this.temperatureType = temperatureType;
   }
 
+  public float getProportion() {
+    return proportion;
+  }
+
+  public void setProportion(float proportion) {
+    this.proportion = proportion;
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/customViews/TemperatureTextView.java]: invalidate or requestLayout

File: app/src/main/java/com/ayush/weatherapp/customViews/TemperatureTextView.java [hunk 4]
```diff
@@ -30,41 +35,61 @@ public TemperatureTextView(Context context, AttributeSet attrs, int defStyleAttr
...
 
     super.setText(spannableText, type);
   }
 
   @NonNull
-  private Spannable getSpannableTextSize(String currentTemp, float proportion, int start, int end) {
+  private Spannable getSpannableTextSize(String currentTemp, int start, int end) {
     Spannable spannableCurrentTemp = new SpannableString(currentTemp);
-    spannableCurrentTemp.setSpan(new RelativeSizeSpan(proportion), start, end,
+    spannableCurrentTemp.setSpan(new RelativeSizeSpan(getProportion()), start, end,
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/customViews/TemperatureTextView.java]: propotion

File: app/src/main/java/com/ayush/weatherapp/home/HomeActivity.java [hunk 5]
```diff
@@ -87,13 +90,19 @@
...
   }
 
   @Override protected void onResume() {
     super.onResume();
     checkLocationPermission();
   }
 
+  @Override protected void onPause() {
+    super.onPause();
+    presenter.onPauseView();
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/home/HomeActivity.java]: presenter.stopLocationUpdate();

File: app/src/main/java/com/ayush/weatherapp/utils/UnitConversionUtils.java [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  public static double fahrenheitToCelsius(double fahrenheit) {
+    return ((fahrenheit - 32) * 5) / 9;
+  }
+
+  public static double celsiusToFahrenheit(double celsius) {
+    return (celsius * 9 / 5) + 32;
+  }
+
+  public static double mphToKmph(double mph) {
+    return (3.6 * mph);
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/utils/UnitConversionUtils.java]: bracket not required

File: app/src/main/java/com/ayush/weatherapp/utils/UnitConversionUtils.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+  }
+
+  public static double kmphToMph(double kmph) {
+    return (0.277778 * kmph);
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/utils/UnitConversionUtils.java]: .

File: app/src/main/java/com/ayush/weatherapp/WeatherApplication.java (hunk 1 above)
Comment (ayush-shta) [reply app/src/main/java/com/ayush/weatherapp/WeatherApplication.java]: :ok:

File: app/src/main/java/com/ayush/weatherapp/constants/Temperature.java (hunk 2 above)
Comment (ayush-shta) [reply app/src/main/java/com/ayush/weatherapp/constants/Temperature.java]: :ok:

File: app/src/main/java/com/ayush/weatherapp/customViews/TemperatureTextView.java (hunk 4 above)
Comment (ayush-shta) [reply app/src/main/java/com/ayush/weatherapp/customViews/TemperatureTextView.java]: :ok:

File: app/src/main/java/com/ayush/weatherapp/customViews/TemperatureTextView.java (hunk 3 above)
Comment (ayush-shta) [reply app/src/main/java/com/ayush/weatherapp/customViews/TemperatureTextView.java]: :ok:

File: app/src/main/java/com/ayush/weatherapp/home/HomeActivity.java (hunk 5 above)
Comment (ayush-shta) [reply app/src/main/java/com/ayush/weatherapp/home/HomeActivity.java]: :ok:

File: app/src/main/java/com/ayush/weatherapp/utils/UnitConversionUtils.java (hunk 6 above)
Comment (ayush-shta) [reply app/src/main/java/com/ayush/weatherapp/utils/UnitConversionUtils.java]: :ok:

File: app/src/main/java/com/ayush/weatherapp/utils/UnitConversionUtils.java (hunk 7 above)
Comment (ayush-shta) [reply app/src/main/java/com/ayush/weatherapp/utils/UnitConversionUtils.java]: :ok:

**Pull request merged**